app/components/avo/fields/common/files/view_type/grid_item_component.html.erb

```diff
@@ -1,5 +1,9 @@
 <div id="<%= dom_id file %>" class="relative min-h-full max-w-full flex-1 flex flex-col justify-between space-y-2">
-  <% if file.present? %>
+  <%# When editing a resource, if you attach a file and also edit another field which causes a validation error.
+      In such a situation, file is an object that hasn't been saved to the database (i.e id == nil) because
+      the resource update failed and so the attachment wasn't persisted.
+      In such situations url_for(file) will complain that it can't find the url for an non-existent file. %>
+  <% if file.persisted? %>
```

Comment:
Is this change still necessary if we always display the persisted image?

Reply:
I think it's better than `file.present?` because it's more clear that the code inside the block needs to execute only if we have a persisted file.

Reply:
Not saying `file.present?` is better, just wondering if this check is necessary or redundant since we already guarantee to show only persisted files

Reply:
True. I removed the conditional. There's a failing test in CI but it passes locally for me.